client.go

```diff
@@ -192,6 +199,10 @@ func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err erro
 	// Otherwise use the client's configured UDP size.
 	if opt == nil && c.UDPSize >= MinMsgSize {
 		co.UDPSize = c.UDPSize
+		// In this case also set the the EDNS0 bufsize to allow for larger responses.
+		if c.SetUDPSize {
+			m.SetEdns0(c.UDPSize, true /* dnssec */)
```

Comment:
I can't decide if using constants or two bools in the struct is better (SetUDPSize, EnableDNSSEC).  The nice part about using struct members is that the code is just
```
if c.SetUDPSize {
  m.SetEDNS0(c.UDPSize, c.EnableDNSSEC)
}
```

Perhaps SetUDPSize is the wrong bool name for the client though?  Is it really SetEDNS0?

At some point the code might need to handle different EDNS versions as well, so maybe an EDNS struct that has {UDPSize, DNSSEC, Version, ...future options...} is a better method to abstract the EDNS options?  Then m.SetEDNS (note not m.SetEDNS0) could take an options struct instead of an argument list.  If this route is selected then the Client struct should have an EDNSOptions member that is passed around instead.

```
type EDNSOptions struct {
  Version int
  UDPSize int
  DNSSEC bool
}

type Client struct {
  ... existing stuff ...
  EDNSOptions *EDNSOptions
  ... existing stuff ...
}

func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err error) {
  ... existing code ...
  if c.EDNSOptions != nil {
    m.SetEDNS(c.EDNSOptions)
  }
  ... existing code...
}

// not exactly sure how this should work, but for discussion...
func (dns *Msg) SetEDNS(opts *EDNSOptions) {
  if opts == nil {
    return
  }
  edns := dns.GetEDNS()
  if edns == nil {
    edns = dns.AddEDNS()  // adds an EDNS OPT RR to dns
  }
  edns.SetVersion(opts.Version)
  edns.SetDNSSec(opts.DNSSEC)
  edns.SetUDPSize(max(MinMsgSize, opts.UDPSize))  
}
```
I haven't placed much thought into what other EDNS options exist to know if this is a valid forward looking solution.